MyMessenger.Angular/src/app/services/auth.service.ts

```diff
@@ -1,43 +1,49 @@
 import { Injectable } from '@angular/core';
+import { Observable, of, BehaviorSubject, map } from 'rxjs';
+import { Response } from './models/response';
 import { HttpClient } from '@angular/common/http';
-import { WrapperService } from './wrapper.service';
-import { Observable, of, BehaviorSubject } from 'rxjs';
-import { Response } from '../models/response';
+import { environment } from '../../environments/environment';
 
 @Injectable({
   providedIn: 'root'
 })
 export class AuthService {
+  private baseUrl = environment.baseUrl;
   private loggedIn = new BehaviorSubject<boolean>(this.isLogged());
   isLoggedIn = this.loggedIn.asObservable();
 
-  constructor(private http: HttpClient, private wrapper: WrapperService) { }
+  constructor(private http: HttpClient) { }
 
-  async login(email: string, password: string): Promise<boolean> {
+  login(email: string, password: string): Observable<boolean> {
     const data = { Email: email, Password: password };
-    const response = await this.wrapper.postAsync('Auth/', data);
-    if (response && response.accessToken && response.refreshToken) {
-      localStorage.setItem('accessToken', response.accessToken);
-      localStorage.setItem('refreshToken', response.refreshToken);
-      this.loggedIn.next(true);
-      return true;
-    } else {
-      console.error('Invalid login credentials.');
-      return false;
+    try{
```

Comment:
In 99% cases you don't need to use `try catch` syntax in Angular, it has it's separate controls to manage this. 
For example here is your refactored code: 

![image](https://github.com/Kuzless/ReenbitMessenger/assets/89141792/b76d375b-09a0-4829-846b-fd1df3b3cb86)

Much more readability

Reply:
merged branch with this change